packages/flutter/lib/src/rendering/editable.dart

```diff
@@ -1507,6 +1527,87 @@ class RenderEditable extends RenderBox with RelayoutWhenSystemFontsChangeMixin {
     }
   }
 
+  @override
+  void assembleSemanticsNode(SemanticsNode node, SemanticsConfiguration config, Iterable<SemanticsNode> children) {
+    assert(_semanticsInfo != null && _semanticsInfo!.isNotEmpty);
+    final List<SemanticsNode> newChildren = <SemanticsNode>[];
+    TextDirection currentDirection = textDirection;
+    Rect currentRect;
+    double ordinal = 0.0;
+    int start = 0;
+    final Queue<SemanticsNode> newChildCache = Queue<SemanticsNode>();
+    for (final InlineSpanSemanticsInformation info in combineSemanticsInfo(_semanticsInfo!)) {
+      assert(!info.isPlaceholder);
+      final TextSelection selection = TextSelection(
+        baseOffset: start,
+        extentOffset: start + info.text.length,
+      );
+      start += info.text.length;
+
+      final TextDirection initialDirection = currentDirection;
+      final List<ui.TextBox> rects = _textPainter.getBoxesForSelection(selection);
+      if (rects.isEmpty) {
+        continue;
+      }
+      Rect rect = rects.first.toRect();
+      currentDirection = rects.first.direction;
+      for (final ui.TextBox textBox in rects.skip(1)) {
+        rect = rect.expandToInclude(textBox.toRect());
+        currentDirection = textBox.direction;
+      }
+      // Any of the text boxes may have had infinite dimensions.
+      // We shouldn't pass infinite dimensions up to the bridges.
+      rect = Rect.fromLTWH(
+        math.max(0.0, rect.left),
+        math.max(0.0, rect.top),
+        math.min(rect.width, constraints.maxWidth),
+        math.min(rect.height, constraints.maxHeight),
+      );
+      // round the current rectangle to make this API testable and add some
+      // padding so that the accessibility rects do not overlap with the text.
+      currentRect = Rect.fromLTRB(
+        rect.left.floorToDouble() - 4.0,
+        rect.top.floorToDouble() - 4.0,
+        rect.right.ceilToDouble() + 4.0,
+        rect.bottom.ceilToDouble() + 4.0,
+      );
+      final SemanticsConfiguration configuration = SemanticsConfiguration()
+        ..sortKey = OrdinalSortKey(ordinal++)
+        ..textDirection = initialDirection
+        ..label = info.semanticsLabel ?? info.text;
+      final GestureRecognizer? recognizer = info.recognizer;
+      if (recognizer != null) {
+        if (recognizer is TapGestureRecognizer) {
+          if (recognizer.onTap != null) {
+            configuration.onTap = recognizer.onTap;
+            configuration.isLink = true;
+          }
+        } else if (recognizer is DoubleTapGestureRecognizer) {
+          if (recognizer.onDoubleTap != null) {
+            configuration.onTap = recognizer.onDoubleTap;
+            configuration.isLink = true;
```

Comment:
Why is isLink true for double tap but not for long press?  Is there always a long press recognizer because that's how you select text?